applications/IgaApplication/custom_utilities/director_utilities.cpp

```diff
@@ -0,0 +1,103 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "director_utilities.h"
+#include "factories/linear_solver_factory.h"
+#include "iga_application_variables.h"
+#include "custom_elements/shell_5p_element.h"
+
+namespace Kratos
+{
+    DirectorUtilities::DirectorUtilities(
+        ModelPart& rModelPart,
+        Parameters JsonParameters)
+        : mrModelPart(rModelPart)
+        , mParameters(JsonParameters)
+    {
+    }
+
+    void DirectorUtilities::ComputeDirectors()
+    {
+        Vector brep_ids = mParameters["brep_ids"].GetVector();
+        for (IndexType i = 0; i < brep_ids.size(); ++i) {
+            auto geometry = mrModelPart.GetRootModelPart().GetGeometry((IndexType)brep_ids[i]);
+
+            size_t number_of_control_points = geometry.size();
+
+            std::vector<IndexType> eqID(number_of_control_points);
+            for (SizeType inodes = 0; inodes < number_of_control_points; ++inodes)
+                eqID[inodes] = geometry[inodes].GetId();
+
+            SparseMatrixType NTN(number_of_control_points, number_of_control_points, number_of_control_points*4); //inital guess how much non-zero are there
+            // Can't we solve each patch independently? -> much more efficient
```

Comment:
If we can, we should definitely do it. having multiple solvers is no issue at all.